producer/src/main/kotlin/streams/procedures/StreamsProcedures.kt

```diff
@@ -15,15 +27,15 @@ class StreamsProcedures {
     @Procedure(mode = Mode.SCHEMA, name = "streams.publish")
     @Description("streams.publish(topic, config) - Allows custom streaming from Neo4j to the configured stream environment")
     fun publish(@Name("topic") topic: String?, @Name("payload") payload: Any?,
-                @Name(value = "config", defaultValue = "{}") config: Map<String, Any>?) {
+                @Name(value = "config", defaultValue = "{}") config: Map<String, Any>?): Stream<PublishResult> {
         checkEnabled()
         if (topic.isNullOrEmpty()) {
             log?.info("Topic empty, no message sent")
```

Comment:
should we rather return this to the client

Reply:
I don't understand this comment.  Do you mean we should not return config, or do you mean we should find a way to return the log message if topic is empty?

Reply:
I meant that no one will see the log. And if we're already returning stuff we should probably return this info (that it was actually not sent instead of an empty stream).

Reply:
You are right -- this is existing code and I would like to keep it to a different PR some other time though for several reasons -- 

When you are sending multiple, it's ambiguous whether you mean "best effort" deliver as many as you can, or if you mean fail if one error happens.  Also if one error happens on number 7, you can't recall 1 - 6.   Error handling here isn't in the scope of the issues I'm trying to tackle, but it is something we should maybe improve.

To overcome the limitations of the semantics of this call, I now have implemented a synchronous send method, which won't succeed unless all are sent, and you get back RecordMetadata.  This is what streams.publish will now use, which has the nice side effect of permitting us to return record metadata from kafka.